test/unit/eigensolver/mc/test_backtransformation.cpp

```diff
@@ -0,0 +1,178 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#include "dlaf/eigensolver/backtransformation.h"
+
+#include <functional>
+#include <sstream>
+#include <tuple>
+#include "gtest/gtest.h"
+#include "dlaf/common/index2d.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/matrix/copy.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/matrix/matrix_base.h"
+#include "dlaf/types.h"
+#include "dlaf/util_matrix.h"
+//#include "dlaf_test/comm_grids/grids_6_ranks.h"
+#include "dlaf_test/matrix/util_matrix.h"
+#include "dlaf_test/matrix/util_matrix_blas.h"
+#include "dlaf_test/matrix/util_matrix_local.h"
+#include "dlaf_test/util_types.h"
+
+using namespace dlaf;
+using namespace dlaf::comm;
+using namespace dlaf::common;
+using namespace dlaf::matrix;
+using namespace dlaf::matrix::internal;
+using namespace dlaf::matrix::test;
+using namespace dlaf::test;
+using namespace testing;
+
+//::testing::Environment* const comm_grids_env =
+//    ::testing::AddGlobalTestEnvironment(new CommunicatorGrid6RanksEnvironment);
+
+template <typename Type>
+class BackTransformationEigenSolverLocalTest : public ::testing::Test {};
+TYPED_TEST_SUITE(BackTransformationEigenSolverLocalTest, MatrixElementTypes);
+
+// template <typename Type>
+// class BackTransformationEigenSolverDistributedTest : public ::testing::Test {
+// public:
+//  const std::vector<CommunicatorGrid>& commGrids() {
+//    return comm_grids;
+//}
+//};
+// TYPED_TEST_SUITE(BackTransformationEigenSolverDistributedTest, double);
+
+GlobalElementSize globalTestSize(const LocalElementSize& size) {
+  return {size.rows(), size.cols()};
+}
+
+const std::vector<std::tuple<SizeType, SizeType, SizeType, SizeType>> sizes = {
+    {3, 0, 1, 1}, {0, 5, 2, 3},  // m, n = 0
+    {2, 2, 3, 3}, {3, 4, 6, 7},  // m < mb
+    {3, 3, 1, 1}, {4, 4, 2, 2}, {6, 3, 3, 3}, {12, 2, 4, 4}, {12, 24, 3, 3}, {24, 36, 6, 6},
+    //  {5, 8, 3, 2}, // {4, 6, 2, 3}, {5, 5, 2, 3},  //{8, 27, 3, 4}, {15, 34, 4, 6},

```

Comment:
These cases lead to operations between incomplete tiles where sizes are not detected correctly. This problem is addressed in issue #354: it could be solved using the sub-tile object.

Reply:
Please add a comment in the code as well.